gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/CourseGradeBreakdownPanel.java

```diff
@@ -0,0 +1,320 @@
+                    } else {
+                        item.add(new Label("average-label", "-"));
+                    }
+                } else {
+                    // This item is a category
+                    String categoryPointsOrWeight;
+                    if (categoryDefinition.getId() == -1) {
+                        categoryPointsOrWeight = "-";
+                        item.add(new AttributeAppender("style", "background: #e6e6e6;"));
+                    } else {
+                        categoryPointsOrWeight = CourseGradeBreakdownPanel.this.weightedCategories ? FormatHelper.formatDoubleAsPercentage(gbItem.getCategoryPointsOrWeight() * 100) : FormatHelper.formatDoubleToDecimal(gbItem.getCategoryPointsOrWeight());
+                        String categoryColor = settings.getCategoryColor(categoryDefinition.getName());
+                        item.add(new AttributeAppender("style", "background: " + categoryColor));
+                        if (categoryDefinition.getExtraCredit()) {
+                            categoryPointsOrWeight = "+" + categoryPointsOrWeight;
+                        }
+                    }
+                    item.add(new Label("out-of-label", categoryPointsOrWeight));
+                    if (!categoryDefinition.getName().equals("Uncategorized")) {
+                        item.add(new Label("average-label", constructCategoryAverageLabel(categoryDefinition)));
+                    } else {
+                        item.add(new Label("average-label", ""));
+                    }
+                    item.add(new AttributeAppender("class", "categoryRow"));
+                }
+            }
+        };
+        itemListContainer.add(itemListView);
+        WebMarkupContainer totalPtsContainer = new WebMarkupContainer("total-points-container") {
+            @Override
+            public boolean isVisible() {
+                return !CourseGradeBreakdownPanel.this.weightedCategories;
+            }
+        };
+        itemListContainer.add(totalPtsContainer);
+        totalPtsContainer.add(new Label("total-points", CourseGradeBreakdownPanel.this.overAllPoints));
+        add(new GbAjaxLink("done") {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public void onClick(final AjaxRequestTarget target) {
+                CourseGradeBreakdownPanel.this.window.close(target);
+            }
+        });
+    }
+
+    private int countNumberofGradedAssignments(List<GbStudentGradeInfo> grades, long assignmentId){
+        int gradedCounter = 0;
+        for(int q = 0; q < grades.size(); q++){
+            final GbGradeInfo gradeInfo = grades.get(q).getGrades().get(assignmentId);
+            if(gradeInfo != null){
+                if(gradeInfo.getGrade() != null){
+                    gradedCounter++;
+                }
+            }
+        }
+        return gradedCounter;
```

Comment:
```suggestion
        int gradedCounter = 0;
        grades.forEach(g -> {
            final GbGradeInfo gradeInfo = g.getGrades().get(assignmentId);
            if (gradeInfo != null) {
                if (gradeInfo.getGrade() != null) {
                    gradedCounter++;
                }
            }
        });
        return gradedCounter;
```

Reply:
Had to convert the counter to AtomicInteger to make it still work in the labmda [had to be Final for that] but this worked fine otherwise.